### PR TITLE
docs: Update examples on scripts execution

### PR DESCRIPTION
**Pull request by dmitry-kabanov**
This PR updates documentation on how to run scripts with dynamically spawned processes for Open MPI implementation as well as adds an example with using MPI4PY_MAX_WORKERS env variable.

It is based on the discussion https://github.com/open-mpi/ompi/issues/8171

### CODE REVIEW

File: docs/source/usrman/mpi4py.futures.rst [hunk 1]
```diff
@@ -421,18 +421,39 @@ scanline is dumped to disk.
                for line in image:
                    f.write(line)
 
-The recommended way to execute the script is using the :program:`mpiexec`
+The recommended way to execute the script is by using the :program:`mpiexec`
 command specifying one MPI process and (optional but recommended) the desired
-MPI universe size [#]_. ::
+`MPI universe size <https://www.mcs.anl.gov/research/projects/mpi/mpi-standard/mpi-report-2.0/node111.htm>`_,
```
Comment (dalcinl) [inline docs/source/usrman/mpi4py.futures.rst:426]: This is not an official URL to the MPI standard, please remove this change.

Comment (dmitry-kabanov) [reply docs/source/usrman/mpi4py.futures.rst]: done

File: docs/source/usrman/mpi4py.futures.rst [hunk 2]
```diff
@@ -421,18 +421,39 @@ scanline is dumped to disk.
...
+which is provided either by a batch system or is set by the user via
+command-line arguments or environment variables. Below we provide
+examples for MPICH and Open MPI as well as `mpi4py`-specific way [#]_.
+In all of these examples, the :program:`mpiexec` command launches a single MPI
+process (the master) running the Python interpreter and executing the main
+script. When required, :mod:`mpi4py.futures` spawns 16 additional MPI processes
+(the children) to dynamically allocate the pool of workers. The master submits
```
Comment (dalcinl) [inline docs/source/usrman/mpi4py.futures.rst]: If you used master before, then here use workers, not children.

Comment (dmitry-kabanov) [reply docs/source/usrman/mpi4py.futures.rst]: How about this?
When required, :mod:`mpi4py.futures` spawns the pool of 16 worker MPI processes

File: docs/source/usrman/mpi4py.futures.rst [hunk 3]
```diff
@@ -421,18 +421,39 @@ scanline is dumped to disk.
...
+variable ::
+
+  $ MPIEXEC_UNIVERSE_SIZE=17 mpiexec -n 1 python julia.py
+
+In the Open MPI implementation, the MPI universe size can be set via ``-host``
+argument to :program:`mpiexec` ::
+
+  $ mpiexec -n 1 -host <hostname>:17 python julia.py
+
+Another, implementation-independent, way to specify the MPI universe size is to
```
Comment (dalcinl) [inline docs/source/usrman/mpi4py.futures.rst:452]: This is wrong. I warned explicitly about this confusion of concepts.  `MPI4PY_MAX_WORKERS` has nothing to do with the MPI universe size. This env var specifies the number of worker processes that mpi4py.futures spawn. Moreover, if the user sets this variable, the MPI universe size is ignored.

File: docs/source/usrman/mpi4py.futures.rst [hunk 4]
```diff
@@ -421,18 +421,39 @@ scanline is dumped to disk.
...
                    f.write(line)
 
-The recommended way to execute the script is using the :program:`mpiexec`
+The recommended way to execute the script is by using the :program:`mpiexec`
 command specifying one MPI process and (optional but recommended) the desired
-MPI universe size [#]_. ::
+`MPI universe size <https://www.mcs.anl.gov/research/projects/mpi/mpi-standard/mpi-report-2.0/node111.htm>`_,
+which is provided either by a batch system or is set by the user via
+command-line arguments or environment variables. Below we provide
+examples for MPICH and Open MPI as well as `mpi4py`-specific way [#]_.
```
Comment (dalcinl) [inline docs/source/usrman/mpi4py.futures.rst]: And here he have potential confusion. The mpi4py-specific way is unrelated to the MPI universe size concept. This wording does not make it clear. Please remove  `as well as mpi4py-specific way`. Discussing MPI4PY_MAX_WORKERS at the end will still be good, but you have to do a better job of not confusing it with the MPI universe size.

File: docs/source/usrman/mpi4py.futures.rst (hunk 2 above)
Comment (dalcinl) [reply docs/source/usrman/mpi4py.futures.rst]: I liked the clarification between parenthesis, but it should have said `(the workers)`. I would keep the parenthesis, to match the previous style.  _A foolish consistency is the hobgoblin of little minds ..._

File: docs/source/usrman/mpi4py.futures.rst (hunk 4 above)
Comment (dmitry-kabanov) [reply docs/source/usrman/mpi4py.futures.rst]: I've removed the phrase `as well as mpi4py-specific way`.

File: docs/source/usrman/mpi4py.futures.rst (hunk 3 above)
Comment (dmitry-kabanov) [reply docs/source/usrman/mpi4py.futures.rst]: I've replaced `MPI universe size` with `number of workers`.

File: docs/source/usrman/mpi4py.futures.rst [hunk 5]
```diff
@@ -421,18 +421,39 @@ scanline is dumped to disk.
...
+argument to :program:`mpiexec` ::
+
+  $ mpiexec -n 1 -host <hostname>:17 python julia.py
+
+Another way to specify the number of workers is to
+use `mpi4py`-specific environment variable::
+
+  $ MPI4PY_MAX_WORKERS=16 mpiexec -n 1 python julia.py
+  
+Notice that in this case MPI universe size is ignored.
```
Comment (dalcinl) [inline docs/source/usrman/mpi4py.futures.rst:456]: Notice or Note? Also: ... **the** MPI universe size ...

File: docs/source/usrman/mpi4py.futures.rst [hunk 6]
```diff
@@ -445,14 +466,9 @@ As explained previously, the 17 processes are partitioned in one master and 16
...
 
-.. [#] This :program:`mpiexec` invocation example using the ``-usize`` flag
-   (alternatively, setting the :envvar:`MPIEXEC_UNIVERSE_SIZE` environment
-   variable) assumes the backend MPI implementation is an MPICH derivative
-   using the Hydra process manager. In the Open MPI implementation, the MPI
-   universe size can be specified by setting the :envvar:`OMPI_UNIVERSE_SIZE`
-   environment variable to a positive integer.  Check the documentation of your
-   actual MPI implementation and/or batch system for the ways to specify the
-   desired MPI universe size.
+.. [#] When using MPI implementations different from MPICH and Open MPI, please
```
Comment (dalcinl) [inline docs/source/usrman/mpi4py.futures.rst:469]: When using **an** MPI implementation **other than** MPICH **or** Open MPI, ... (please note I'm not an English speaker).

File: docs/source/usrman/mpi4py.futures.rst [hunk 7]
```diff
@@ -445,14 +466,9 @@ As explained previously, the 17 processes are partitioned in one master and 16
...
+   check the documentation of your actual MPI implementation and/or batch
```
Comment (dalcinl) [inline docs/source/usrman/mpi4py.futures.rst:470]: And here he have "MPI implementation" again... There is a reason I don't write docs, getting it right is hard.

File: docs/source/usrman/mpi4py.futures.rst [hunk 8]
```diff
@@ -421,18 +421,39 @@ scanline is dumped to disk.
...
+
+  $ MPIEXEC_UNIVERSE_SIZE=17 mpiexec -n 1 python julia.py
+
+In the Open MPI implementation, the MPI universe size can be set via ``-host``
+argument to :program:`mpiexec` ::
+
+  $ mpiexec -n 1 -host <hostname>:17 python julia.py
+
+Another way to specify the number of workers is to
+use `mpi4py`-specific environment variable::
```
Comment (dalcinl) [inline docs/source/usrman/mpi4py.futures.rst:452]: use **a** `mpi4py.futures`-specific variable ...

Or perhaps

set the `MPI4PY_MAX_WORKERS` environment variable

File: docs/source/usrman/mpi4py.futures.rst (hunk 7 above)
Comment (dmitry-kabanov) [reply docs/source/usrman/mpi4py.futures.rst:470]: Do you mean the repetition of "MPI implementation" in the same sentence? I do not see at as a problem.

**Comment by dalcinl** [comment]
@dmitry-kabanov Sorry for the delay doing my final review.  Ready for merge? Please confirm.

**Comment by dmitry-kabanov** [comment]
Yes, I think so

**Comment by dalcinl** [comment]
@dmitry-kabanov I'm not sure if the squash-and-merge option will preserve your authorship. Could you please fixup all your commits in a single one?

**Comment by dalcinl** [comment]
Sorry, after reading the docs, I can just squash and merge. Doing it now.

**Comment by dmitry-kabanov** [comment]
Great, thanks!

**Pull request merged**